connectors/sources/mssql.py

```diff
@@ -73,42 +81,67 @@ def get_default_configuration(cls):
         mssql_configuration = super().get_default_configuration().copy()
         mssql_configuration.update(
             {
-                "schema": {"value": "dbo", "label": "Schema", "type": "str"},
-                "mssql_driver": {
-                    "value": "ODBC Driver 18 for SQL Server",
-                    "label": "Microsoft SQL Driver Name (ODBC Driver 18 for SQL Server)",
+                "schema": {
+                    "label": "Schema",
+                    "type": "str",
+                    "value": "dbo",
+                },
+                "ssl_enabled": {
+                    "label": "Enable SSL verification (true/false)",
+                    "type": "bool",
+                    "value": False,
+                },
+                "ssl_ca": {
+                    "label": "Certificate Data",
                     "type": "str",
+                    "value": "",
                 },
-                "secured_connection": {
-                    "value": SECURED_CONNECTION,
-                    "label": "Connection will be secured or not",
+                "validate_host": {
+                    "label": "Do you want to validate host",
                     "type": "bool",
+                    "value": False,
                 },
             }
         )
         return mssql_configuration
 
     def _create_engine(self):
         """Create sync engine for mssql"""
-        if self.secured_connection:
-            query = {
-                "driver": self.mssql_driver,
-                "TrustServerCertificate": "no",
-                "Encrypt": "Yes",
-            }
-        else:
-            query = {"driver": self.mssql_driver, "TrustServerCertificate": "yes"}
-
         connection_string = URL.create(
-            "mssql+pyodbc",
+            "mssql+pytds",
             username=self.user,
             password=self.password,
             host=self.host,
             port=self.port,
             database=self.database,
-            query=query,
         )
-        self.engine = create_engine(connection_string)
+        connect_args = {}
+        if self.ssl_enabled:
+            self.create_pem_file()
+            connect_args = {
+                "cafile": f"/tmp/{self.certfile}",
+                "validate_host": self.validate_host,
+            }
+        self.engine = create_engine(connection_string, connect_args=connect_args)
+
+    async def close(self):
+        """Close the connection to the database server."""
+        if os.path.exists(f"/tmp/{self.certfile}"):
+            try:
+                os.remove(f"/tmp/{self.certfile}")
+            except Exception as exception:
+                logger.warning(
+                    f"Something went wrong while removing temporary certificate file. Exception: {exception}"
+                )
+        if self.connection is None:
+            return
+        self.connection.close()
+
+    def create_pem_file(self):
```

Comment:
use a named temp file, and not an explicit `/tmp/` path, and have that function return the file path, you can store for deletion

Reply:
added for a follow up fix https://github.com/elastic/connectors-python/issues/764

Reply:
Sure, We'll raise follow-up PR for #764. I think we can land this backport PR for now.